fix(api): db write optimization

amitlissack opened the pull request:
# Overview

Database migration takes a very long time. This attempts to reduce that time by reducing the number of inserts we make.

# Changelog

- added insert_wells_into_db that uses `executemany` to insert a number of rows vs just one.

# Review requests

# How I tested

Ran the the full migration on `edge` and in `api_db_speed`.  Then compared the two databases (steps 9 and 12 should show no differences in the files).  Also, observing that robot-server takes longer to be ready in step 3 than step 6.

```
1. git checkout edge
2. rm ~/.opentrons/opentrons.db
3. make -C robot-server dev
4. mv ~/.opentrons/opentrons.db ~/.opentrons/opentrons_orig.db 
5. git checkout api_db_speed
6. make -C robot-server dev
7. sqlite3 ~/.opentrons/opentrons_orig.db "select * from Containers order by name" > orig.txt
8. sqlite3 ~/.opentrons/opentrons.db "select * from Containers order by name" > new.txt
9. diff new.txt orig.txt
10. sqlite3 ~/.opentrons/opentrons.db "select * from ContainerWells order by container_name" > new.txt
11. sqlite3 ~/.opentrons/opentrons_orig.db "select * from ContainerWells order by container_name" > orig.txt
12. diff orig.txt new.txt

```

# Risk assessment

Low. Only affects database migration.


closes #6029 , #5983

Commit:
insert all container wells for a container in one executemany call

Commit:
teeny optimization

Commit:
minor clean up

sfoster1 reviewed (approved):
LGTM! Thanks!

Laura-Danielle commented:
I had trouble testing this on my robot, trying a different method to see if that works. (When I removed the database it killed the server and never came back up)

amitlissack commented:
> I had trouble testing this on my robot, trying a different method to see if that works. (When I removed the database it killed the server and never came back up)

mcous reviewed (approved):
Completely unscientific tests seem to indicate a ~50% reduction in boot time on my robot after the DB has been cleared. Procedure:

1. Push branch using `make push`
2. Reset labware calibration via app
3. Start timer on "Restart" click

branch  | time to pipette home | time to app discoverable
--- | --- | ---
`api_db_speed` | 50 s | 120 s
`edge` | 105 s | 130 s

amitlissack commented:
> Completely unscientific tests seem to indicate a ~50% reduction in boot time on my robot after the DB has been cleared. Procedure:
> 
>     1. Push branch using `make push`
> 
>     2. Reset labware calibration via app
> 
>     3. Start timer on "Restart" click
> 
> 
> branch 	time to pipette home 	time to app discoverable
> `api_db_speed` 	50 s 	120 s
> `edge` 	105 s 	130 s

That is roughly what I saw. Building `opentrons.db` went from 0:59 to 0:07.

amitlissack commented:
> I had trouble testing this on my robot, trying a different method to see if that works. (When I removed the database it killed the server and never came back up)

Sorry about that. That was due to the Pydantic issue I fixed in another PR. I rebased this branch with that fix.

Laura-Danielle reviewed (approved):
Works for me now 👍
